workers/unity/Assets/Fps/Scripts/SetupLogic/ConnectionController.cs

```diff
@@ -12,93 +13,87 @@ public class ConnectionController : MonoBehaviour
         [Require] private PlayerCreator.Requirable.CommandRequestSender commandSender;
         [Require] private PlayerCreator.Requirable.CommandResponseHandler responseHandler;
 
-        private GameObject canvasCameraObj;
-        private ScreenUIController screenUIController;
-        private Animator connectButton;
         private WorkerConnector clientWorkerConnector;
 
         private void Start()
         {
+            ConnectionStateReporter.InformOfConnectionController(this);
             clientWorkerConnector = gameObject.GetComponent<WorkerConnector>();
-            connectButton = screenUIController.ConnectScreen.GetComponentInChildren<Animator>();
+            clientWorkerConnector.OnWorkerCreationFinished += OnWorkerCreated;
         }
 
-        private void OnEnable()
+        private void OnWorkerCreated(Worker worker)
         {
-            if (responseHandler != null)
+            if (worker.Connection.GetConnectionStatusCode() == ConnectionStatusCode.Success)
             {
-                responseHandler.OnCreatePlayerResponse += OnCreatePlayerResponse;
+                StartCoroutine(nameof(DelayedConnectedMessage));
+            }
+            else
+            {
+                ConnectionStateReporter.SetState(ConnectionStateReporter.State.ConnectionFailed,
+                    worker.Connection.GetConnectionStatusCode().ToString());
             }
         }
 
-        public void InformOfUI(GameObject canvasCameraObj, ScreenUIController screenUIController)
+        private IEnumerator DelayedConnectedMessage()
         {
-            this.canvasCameraObj = canvasCameraObj;
-            this.screenUIController = screenUIController;
+            // 1 frame delay necessary to allow [Require] components to active on ConnectionController
+            yield return null;
+            ConnectionStateReporter.SetState(ConnectionStateReporter.State.Connected);
         }
 
-        private void OnCreatePlayerResponse(PlayerCreator.CreatePlayer.ReceivedResponse obj)
+        private void OnEnable()
         {
-            if (obj.StatusCode == StatusCode.Success)
-            {
-                canvasCameraObj.SetActive(false);
-                screenUIController.ConnectScreen.SetActive(false);
-                screenUIController.InGameHud.SetActive(true);
-            }
-            else
+            if (responseHandler != null)
```

Comment:
This null check shouldn't be necessary as its a broken contract for `Require`

Reply:
I'm not comfortable making changes to connection scripts in this PR, which is intended to be UI focused. This logic is all present in /develop branch already. If this is just an observation...it might have better visibility elsewhere? (Honestly not sure, I'm still kinda green to this whole github business)